xapian-core/backends/dbfactory.cc

```diff
@@ -360,7 +360,7 @@ open_stub(WritableDatabase &db, const string &file, int flags)
     }
 
     if (db.internal->size() == 0) {
-	throw DatabaseOpeningError(file + ": No databases listed");
+	throw DatabaseNotFoundError(file + ": No databases listed");
```

Comment:
I'm not sure which exception is more appropriate here - this is for opening a "stub database file", which is a text file which should list one or more databases to open.  In this case the stub file was found but no databases were listed in it.

@Kronuz Does the behaviour in this situation matter for your use case?

Reply:
I think it should be `DatabaseOpeningError`:

1. Pretty sure you can't use it to create a new database as @Kronuz would need for `DatabaseNotFoundError`;
2. It feels to me more like an invalid database than a missing one.

Reply:
> Are there any other changes that you would suggest?

As discussed above, I think this one should stay as `DatabaseOpeningError`.

(Perhaps this case shouldn't actually be an error, but should instead just open a database with zero shards - i.e. an empty database; currently to specify an empty database in a stub file you have to put `inmemory`.  But if we change that it should really be done separately to the changes for adding `DatabaseNotFoundError`).

Reply:
> I think both these cases should throw `DatabaseNotFoundError` (and Kronuz agrees).
>
I thought you were implying `DatabaseNotFoundError` but as you mentioned that keeping `DatabaseNotFoundError` would require furthur changes. Thank you for suggesting the changes. I will push the changes.

Reply:
sorry i didn't notice.thanks fixed.